src/test/java/org/java_websocket/issues/Issue997Test.java

```diff
@@ -0,0 +1,174 @@
+        SSLWebSocketClient client = testIssueWithLocalServer("localhost", SocketUtil.getAvailablePort(), SSLContextUtil.getLocalhostOnlyContext(), SSLContextUtil.getLocalhostOnlyContext(), "HTTPS");
+        assertTrue(client.onOpen);
+        assertFalse(client.onSSLError);
+    }
+    @Test(timeout=2000)
+    public void test_localServer_ServerLocalhost_ClientLocalhost_CheckInactive() throws CertificateException, UnrecoverableKeyException, NoSuchAlgorithmException, KeyManagementException, KeyStoreException, IOException, URISyntaxException, InterruptedException {
+        SSLWebSocketClient client = testIssueWithLocalServer("localhost", SocketUtil.getAvailablePort(), SSLContextUtil.getLocalhostOnlyContext(), SSLContextUtil.getLocalhostOnlyContext(), "");
+        assertTrue(client.onOpen);
+        assertFalse(client.onSSLError);
+    }
+
+
+    public SSLWebSocketClient testIssueWithLocalServer(String address, int port, SSLContext serverContext, SSLContext clientContext, String endpointIdentificationAlgorithm) throws IOException, URISyntaxException, InterruptedException {
+        CountDownLatch countServerDownLatch = new CountDownLatch(1);
+        SSLWebSocketClient client = new SSLWebSocketClient(address, port, endpointIdentificationAlgorithm);
+        WebSocketServer server = new SSLWebSocketServer(port, countServerDownLatch);
+
+        server.setWebSocketFactory(new DefaultSSLWebSocketServerFactory(serverContext));
+        if (clientContext != null) {
+            client.setSocketFactory(clientContext.getSocketFactory());
+        }
+        server.start();
+        countServerDownLatch.await();
+        client.connectBlocking(1, TimeUnit.SECONDS);
+        return client;
+    }
+
+
+    private static class SSLWebSocketClient extends WebSocketClient {
+        private final String endpointIdentificationAlgorithm;
+        public boolean onSSLError = false;
+        public boolean onOpen = false;
+
+        public SSLWebSocketClient(String address, int port, String endpointIdentificationAlgorithm) throws URISyntaxException {
+            super(new URI("wss://"+ address + ':' +port));
+            this.endpointIdentificationAlgorithm = endpointIdentificationAlgorithm;
+        }
+
+        @Override
+        public void onOpen(ServerHandshake handshakedata) {
+            this.onOpen = true;
+        }
+
+        @Override
+        public void onMessage(String message) {
+        }
+
+        @Override
+        public void onClose(int code, String reason, boolean remote) {
+        }
+
+        @Override
+        public void onError(Exception ex) {
+            if (ex instanceof SSLHandshakeException) {
+                this.onSSLError = true;
+            }
+        }
+
+        @Override
+        protected void onSetSSLParameters(SSLParameters sslParameters) {
+            if (endpointIdentificationAlgorithm == null) {
```

Comment:
If I understand correctly, endpointIdentificationAlgorithm is never actually null in this test. An empty string is used instead.

Reply:
Yeah you are right. I started out with NULL but just went to an empty string